controllers/storageconfig_controller.go

```diff
@@ -92,12 +92,15 @@ func (r *StorageSecretReconciler) reconcileSecret(secret *corev1.Secret,
 	}
 	err := r.List(ctx, dataConnectionSecretsList, opts...)
 	if err != nil {
-		if apierrs.IsNotFound(err) {
```

Comment:
> Said differently can you confirm that the error was not produced in case of no data connection?
Correct. There is no error thrown even if there is no data connection. It returns just nil. 
Upon reflection, it seems appropriate that no error is thrown because this isn't about finding a specific secret by name. Additionally, I've already tested it manually.